src/librustc_trans/consts.rs

```diff
@@ -42,6 +43,25 @@ pub fn bitcast(val: ValueRef, ty: Type) -> ValueRef {
     }
 }
 
+fn set_global_alignment(ccx: &CrateContext,
+                        gv: ValueRef,
+                        mut align: machine::llalign) {
+    // The target may require greater alignment for globals than the type does.
+    // Note: GCC and Clang also allow `__attribute__((aligned))` on variables,
+    // which can force it to be smaller.  Rust doesn't support this yet.
+    if let Some(min) = ccx.sess().target.target.options.min_global_align {
+        match ty::layout::Align::from_bits(min, min) {
+            Ok(min) => align = cmp::max(align, min.abi() as machine::llalign),
```

Comment:
There's a `.min` method on `Align`, you should use that.

Reply:
We want the greater of the two, so it would be `Align::max`. But this would require converting the incoming `machine::llalign` to `Align`, comparing, then back to `llalign` for the LLVM call. I don't see the point of that.

Reply:
Well, `llalign` should have never existed etc. etc. I'll just have to remember fixing it on my branch where it's gone, but the rebase should catch it because of type mismatches and whatnot.

Reply:
Maybe it's also pointless to convert `min_global_align` to `Align`. I just figured a little sanity checking of that value would be good.

Reply:
That conversion is fine, since it will be in there sooner or later! In the future I plan to have `librustc_back` and a lot of `librustc/ty/layout.rs` be in the same crate.